java-checks/src/main/java/org/sonar/java/checks/KnownCapacityHashBasedCollectionCheck.java

```diff
@@ -61,7 +61,7 @@ protected MethodMatchers getMethodInvocationMatchers() {
 
   @Override
   public boolean isCompatibleWithJavaVersion(JavaVersion version) {
-    return version.isJava19Compatible();
+    return version.isJava20Compatible();
```

Comment:
We should not change the above line. This rule tries to enforce new Java methods introduced in Java 19, not 20.
Like [HashMap#newHashMap(int)](https://docs.oracle.com/en/java/javase/19/docs/api/java.base/java/util/HashMap.html#newHashMap(int))